### PR TITLE
add pretty print to object mod

### PR DESCRIPTION
**Pull request by briandowns**
Resolves: #682 

### What's Changed:

# Added pretty print function into the Object module. 

### Type of Change:

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#

### Housekeeping:

- [x] Tests have been updated to reflect the changes done within this PR (if applicable).
- [x] Documentation has been updated to reflect the changes done within this PR (if applicable).

#

### CODE REVIEW

File: docs/docs/standard-lib/object.md [hunk 1]
```diff
@@ -61,3 +61,27 @@ This method will return a string of the object's hash value.
 ```cs
 Object.hash("Dictu");
 ```
+
+### Object.prettyPrint(Value, indent -> Optional) -> Result\<String>
+
+This method will return a string representation of the given value.
+
+**NOTE** Strings, dicts, lists, numbers, and nil are valid values for pretty printing at this time.
```
Comment (Jason2605) [inline docs/docs/standard-lib/object.md:69]:
```suggestion
**NOTE** Strings, dicts, lists, numbers, booleans and nil are valid values for pretty printing at this time.
```

File: docs/docs/standard-lib/object.md [hunk 2]
```diff
@@ -61,3 +61,27 @@ This method will return a string of the object's hash value.
 ```cs
 Object.hash("Dictu");
 ```
+
+### Object.prettyPrint(Value, indent -> Optional) -> Result\<String>
```
Comment (Jason2605) [inline docs/docs/standard-lib/object.md:65]:
```suggestion
### Object.prettyPrint(Value, Number: indent -> Optional) -> Result\<String>
```

File: docs/docs/standard-lib/object.md [hunk 3]
```diff
@@ -61,3 +61,27 @@ This method will return a string of the object's hash value.
...
+
+```cs
+Object.prettyPrint([1, 2, 3]).unwrap();
+
+// Output
+'[1, 2, 3]'
+```
+
+```cs
+Object.prettyPrint({"a": 1}, 4).unwrap()
```
Comment (Jason2605) [inline docs/docs/standard-lib/object.md:79]:
```suggestion
Object.prettyPrint({"a": 1}, 4).unwrap();
```

File: src/optionals/object/object.c [hunk 4]
```diff
@@ -59,7 +59,21 @@ static Value objectHash(DictuVM *vm, int argCount, Value *args) {
...
+
+    return OBJ_VAL(stringify(vm, argCount, args));
+}
+
 Value createObjectModule(DictuVM *vm) {
+    ObjString *name = copyString(vm, "Object", 6);
+    push(vm, OBJ_VAL(name));
+    ObjModule *module = newModule(vm, name);
+    push(vm, OBJ_VAL(module));
+
```
Comment (Jason2605) [inline src/optionals/object/object.c:76]: We don't need any of this, we have access to the module through the closure

File: src/optionals/object/object.c [hunk 5]
```diff
@@ -75,6 +89,8 @@ Value createObjectModule(DictuVM *vm) {
     defineNative(vm, &closure->function->module->values, "getClassRef", objectGetClassRef);
     defineNative(vm, &closure->function->module->values, "hash", objectHash);
 
+    defineNative(vm, &module->values, "prettyPrint", objectPrettyPrint);
```
Comment (Jason2605) [inline src/optionals/object/object.c:92]:
```suggestion
    defineNative(vm, &closure->function->module->values, "prettyPrint", objectPrettyPrint);
```

**Review by Jason2605 — COMMENTED** [review]
Couple of comments!

**Comment by Jason2605** [comment]
A question for you, would you expect a method called `prettyPrint` to actually output to stdout rather than return a string? I think a method that returns the pretty output as a string would be useful but potentially under a different name, what do you think?

**Comment by briandowns** [comment]
Yeah, that's a good point. I'll update this to print to stdout. What would you want the method name to be that would returned the "pretty" string?

**Comment by Jason2605** [comment]
Actually we probably don't need the non-print version as you can just use `JSON.stringify` - at least for the minute until we extend it so we can pretty print more objects that aren't JSON serialisable

**Comment by briandowns** [comment]
Good point.

**Comment by Jason2605** [comment]
Thanks so much for this!

**Pull request merged**